### PR TITLE
Fix docstrings, typing and string construction - DM

### LINKED ISSUE #94: Fix docstrings, typing and string construction - DM

### PR DESCRIPTION
**Pull request by NoB0**
Fixes #94

### CODE REVIEW

File: moviebot/dialogue_manager/dialogue_context.py [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
-        self.movies_recommended = {}
-        self.previous_utterances = []
-        # self.movies_recommended.update(previous_recommendation)
-
-    def add_utterance(self, utterance):
         self.previous_utterances.append(utterance)
 
-    def __str__(self):
+    def __str__(self) -> str:
+        """Returns the string representation of the movie recommended."""
```
Comment (IKostric) [inline moviebot/dialogue_manager/dialogue_context.py:32]: Nit: movies

I believe all recomended movies in the conversation are stored in this dictionary.

File: moviebot/dialogue_manager/dialogue_state.py [hunk 2]
```diff
@@ -1,29 +1,23 @@
...
-    It is a basic SlotFilling model that keeps account of the question
-    agents must answer or ask from the user. It also keeps track of the
-    recommendation agent makes and previous states of the agent. State
-    will be updated using the Dialogue State Tracker.
-    """
-
-    def __init__(self, ontology: Ontology, slots, isBot):
-        """Initializes the Slot Filling Dialogue State structures.
+    def __init__(
+        self, ontology: Ontology, slots: Iterable, isBot: bool
```
Comment (IKostric) [inline moviebot/dialogue_manager/dialogue_state.py:18]: Nit: slots is a list of string.

File: moviebot/dialogue_manager/dialogue_state_tracker.py [hunk 3]
```diff
@@ -323,13 +288,16 @@ def update_state_agent(self, agent_dacts):
                     self.ontology.user_requestable
                 )
 
-    def update_state_db(self, database_result=None, backup_results=None):
-        """Updates the state based on  the results fetched from the database.
+    def update_state_db(
+        self, database_result: List = None, backup_results: List = None
```
Comment (IKostric) [inline moviebot/dialogue_manager/dialogue_state_tracker.py:292]: Nit: List[Dict[str, Any]]

If possible I like to have at least one level in on lists and dictionaries.

**Review by WerLaj — APPROVED** [review]
LGTM, several minor issues mentioned in the comments

File: moviebot/dialogue_manager/dialogue_act.py [hunk 4]
```diff
@@ -1,4 +1,7 @@
 """Dialogue Act defines the action user or agent takes during the conversation.
+
+The DialogueAct comprises of an intent with a list of parameters
```
[the remark this answers is not in the document]
Comment (NoB0) [reply moviebot/dialogue_manager/dialogue_act.py:3]: Done.

File: moviebot/dialogue_manager/dialogue_act.py [hunk 5]
```diff
@@ -7,16 +10,12 @@
 
 
 class DialogueAct:
-    """The DialogueAct comprises of an intent with a list of parameters
-    (DialogueActItem) for a particular dialogue."""
-
-    def __init__(self, intent=None, params=None):
-        """Initialises a Dialogue Act.
+    def __init__(self, intent=None, params=None) -> None:
```
[the remark this answers is not in the document]
Comment (NoB0) [reply moviebot/dialogue_manager/dialogue_act.py:13]: Done.

File: moviebot/dialogue_manager/dialogue_context.py (hunk 1 above)
Comment (NoB0) [reply moviebot/dialogue_manager/dialogue_context.py:32]: Done.

File: moviebot/dialogue_manager/dialogue_manager.py [hunk 6]
```diff
@@ -109,7 +106,7 @@ def generate_output(self, restart: bool = False):
 
         return agent_dacts
 
-    def database_lookup(self):
+    def database_lookup(self) -> List:
```
[the remark this answers is not in the document]
Comment (NoB0) [reply moviebot/dialogue_manager/dialogue_manager.py:109]: Done.

File: moviebot/dialogue_manager/dialogue_policy.py [hunk 7]
```diff
@@ -262,19 +266,21 @@ def next_action(  # noqa: C901
                     )
         return agent_dacts
 
-    def _generate_examples(self, database_result, slot):
-        """
+    def _generate_examples(self, database_result: List, slot: str) -> str:
```
[the remark this answers is not in the document]
Comment (NoB0) [reply moviebot/dialogue_manager/dialogue_policy.py:269]: Done.

File: moviebot/dialogue_manager/dialogue_state.py (hunk 2 above)
Comment (NoB0) [reply moviebot/dialogue_manager/dialogue_state.py:18]: Done.

File: moviebot/dialogue_manager/dialogue_state_tracker.py (hunk 3 above)
Comment (NoB0) [reply moviebot/dialogue_manager/dialogue_state_tracker.py:292]: Done.

File: moviebot/dialogue_manager/dialogue_state_tracker.py [hunk 8]
```diff
@@ -13,9 +15,7 @@
 
 
 class DialogueStateTracker:
-    """Dialogue State Tracker updates the current dialogue state."""
-
-    def __init__(self, config, isBot: bool):
+    def __init__(self, config: Dict, isBot: bool) -> None:
```
[the remark this answers is not in the document]
Comment (NoB0) [reply moviebot/dialogue_manager/dialogue_state_tracker.py:18]: Done.

**Pull request merged**